sdk/src/Security/Impl/Ed25519Signer.cs

```diff
@@ -27,8 +30,10 @@ public string Sign(IMessage message)
 
         public string Sign(string payload)
         {
-            var signature = PublicKeyAuth.SignDetached(payload, privateKey);
-            return Base64UrlEncoder.Encode(signature);
+            var payloadBytes = Encoding.UTF8.GetBytes(payload);
+            signer.Reset();
```

Comment:
I'm not familiar with C# but this looks like it might make the class not thread-safe. Should this method by synchronized somehow or should we create a new Signer for each request?

Reply:
To be fair we don't if the old library call was thread-safe either, does that we need to keep in mind to check whether library calls are thread-safe? but I agree better safe than sorry, added the "Synchronized" keyword for both methods.

Reply:
Yeah we should always take care to avoid or synchronize libraries that may be thread unsafe (or document the classes that use them as unsafe themselves). Usually it's pretty obvious - I know in Java classes are usually annotated or documented as such, not sure what the convention is in C# - and in cases like this the api makes it pretty clear that there would be some synchronization concerns if multiple threads were to share this object.